api.py

```diff
@@ -0,0 +1,76 @@
+import os
+import logging
+import requests
+import flask
+import datetime
+
+
+from flask import Flask, request, flash, redirect, render_template, url_for, jsonify
+from flask_sqlalchemy import SQLAlchemy
+
+app = Flask(__name__)
+app.config.update(dict(
+    SQLALCHEMY_DATABASE_URI='sqlite:///kraken.db',
+    DEBUG=True,
+    SECRET_KEY='development',
+    LOG_LEVEL=logging.DEBUG,
+    ENVIRONMENT='development'
+))
+app.config.from_envvar('FLASK_SETTINGS', silent=True)
+
+logging.basicConfig(level=logging.DEBUG, filename='logs/' + app.config['ENVIRONMENT'] + '.log')
+
+db = SQLAlchemy(app)
+
+# TODO: reused model from sesh-dash! DRY!!! - at least move database stuff it into a separate file
+class Sesh_Site(db.Model):
+    __tablename__= 'sites'
+    id = db.Column(db.Integer, primary_key=True)
+    site_name = db.Column(db.String(100))
+
+    def __str__(self):
+        return self.site_name
+
+class BoM_Data_Point(db.Model):
```

Comment:
yes, that is what I mean. perfect, will do that. I think that is a good
compromise between DRY and little effort.

have you done the loading of the schema with sqlalchemy already? I failed
to do it so I added the column definitions here. if not, I will figure it
out :)

thanks.




On Wednesday, December 16, 2015, Alp Tilev <notifications@github.com> wrote:

> In api.py
> <https://github.com/GreatLakesEnergy/sesh-api-helper/pull/1#discussion_r47772207>
> :
>
> > +app.config.from_envvar('FLASK_SETTINGS', silent=True)
> > +
> > +logging.basicConfig(level=logging.DEBUG, filename='logs/' + app.config['ENVIRONMENT'] + '.log')
> > +
> > +db = SQLAlchemy(app)
> > +
> > +# TODO: reused model from sesh-dash! DRY!!! - at least move database stuff it into a separate file
> > +class Sesh_Site(db.Model):
> > +    __tablename__= 'sites'
> > +    id = db.Column(db.Integer, primary_key=True)
> > +    site_name = db.Column(db.String(100))
> > +
> > +    def __str__(self):
> > +        return self.site_name
> > +
> > +class BoM_Data_Point(db.Model):
>
> What if we do this. We point the FLASK app towards the SESH_DB database
> and it extracts the tables and data types straight form the DB. So Django
> governs the DB schema. This app shouldn't govern the schema but simply
> check and write to what is there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/GreatLakesEnergy/sesh-api-helper/pull/1/files#r47772207>
> .
>


-- 
Michael Bumann

F       +49 (0) 176 22747893
E       hello@michaelbumann.com
T       twitter.com/bumi